### PR TITLE
Feat/show questions and answer possibilities

### LINKED ISSUE #1: Set up questions and answer possibilities
## Value statement 
As a club founder
I need bylaws for my club (required by law). I want to answer simple questions
so that I can get individual bylaws for my club. 

## Description (Text, Scribble, Wireframe, Design)
Image from iOS (10)
Image from iOS (11)
Image from iOS (12)
Image from iOS (13)

## Acceptance criteria
- [x] User can answer all questions / by clicking or using inputs (no validation of inputs)
- [x] User can get an overview of his answers by clicking "Zusammenfassung" 

## Tasks
- [x] create feature branch
- [x] initialize react-app
- [x] remove unnecessary template files
- [x] research suitable questions and answers for creating bylaws 
- [x] create header
- [x] add questions as json file
- [x] create questions
- [x] create answer possibilities and inputs
- [x] show inputs/answers when button "Zusammenfassung" is clicked
- [x] add GlobalStyles
- [x] add components in storybook
- [x] unit testing

## Size
Medium/Large

### LINKED ISSUE #4: Basic Design
## Value statement - a requirement, not a solution
As a developer
I need to layout my app
so that a user might like the app and use it

## Description (Text, Scribble, Wireframe, Design)
- simple design
- preferred color-schemes: 
Screenshot 2020-08-26 at 17 03 55

or 
Screenshot 2020-08-28 at 12 38 24

- font ideas: 
   - Header: https://fonts.google.com/specimen/Playfair+Display?category=Serif,Monospace
   - Text: https://fonts.google.com/specimen/Lato?query=lato

## Acceptance criteria
- [x] looks ok

## Tasks
- [x] create design in Adobe XD 
- [x] add fonts
- [x] add colors
- [x] adjust margin/padding
- [x] style buttons 
- [x] style header

## Size
small

### PR DESCRIPTION
**Pull request by carolinwagner**
Please review. 
Preview deployment: https://capstone-proje-git-feat-show-questions-and-answer-possib-96041b.carolinwagner.vercel.app/

Closes #1 and closes #4

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -9,6 +9,10 @@
       content="Web site created using create-react-app"
     />
     <title>hello bylaws</title>
+    <link
+      href="https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;0,900;1,100;1,300;1,400;1,700;1,900&family=Playfair+Display:ital,wght@0,400;0,500;0,600;0,669;0,700;0,800;0,900;1,400;1,500;1,600;1,669;1,700;1,800;1,900&display=swap"
```
Comment (karstenlenz) [inline public/index.html:13]: You are loading a lot of different font-weights. Do you really need all of them? If not you should clean up later.

File: src/App.js [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
+
+  return (
+    <>
+      <Header />
+      <QuestionsForm onClick={addAnswers} />
+      <Summary answers={answers} />
+    </>
+  )
+
+  function addAnswers(newAnswers) {
```
Comment (karstenlenz) [inline src/App.js:17]: Do you need this function? It seems like it only calls the setAnswers function. In that case I think you could directly pass setAnswers as the prop.

File: src/Button.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import styled from 'styled-components/macro'
+
+const StyledButton = styled.button`
+  padding: 20px;
+  color: var(--lightgrey);
```
Comment (karstenlenz) [inline src/Button.js:5]: I like that you use CSS variables for colors!

File: src/Button.test.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import '@testing-library/jest-dom/extend-expect'
+import { render } from '@testing-library/react'
+import Button from './Button'
+
+describe('Button', () => {
+  it('renders Button correctly', () => {
```
Comment (karstenlenz) [inline src/Button.test.js:7]: You dont't need to write Button, you can just say: `it('renders correctly, ()={...}`
Jest will use the "describe"-part and replace the "it" with it. Have a look at the test results, I hope you can understand what I mean.

File: src/GlobalStyles.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+body {
+  font-size: 112.5%;
+  font-family: 'Lato', sans-serif;
+  margin: 50px;
+}
+
+:root {
+    --black-main: #292424;
```
Comment (karstenlenz) [inline src/GlobalStyles.js:16]: Is that really black? The hex code looks grey to me.

File: src/QuestionsForm.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+  const onSubmit = (data) => {
+    console.log(data)
+    onClick(data)
+  }
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      {questions.map((question, index) => (
+        <React.Fragment key={index}>
```
Comment (karstenlenz) [inline src/QuestionsForm.js:22]: Dou you need to write "React.Fragment" or can you just use "<>" ?

**Review by karstenlenz — APPROVED** [review]
Great job! I only left some very minor comments 👍

File: public/index.html (hunk 1 above)
Comment (carolinwagner) [reply public/index.html:13]: Good point - I just linked all fonts in advance. I will clean it up!

File: src/App.js (hunk 2 above)
Comment (carolinwagner) [reply src/App.js:17]: True - my thought on it was to improve readability by naming the function "addAnswers"

File: src/GlobalStyles.js (hunk 5 above)
Comment (carolinwagner) [reply src/GlobalStyles.js:16]: I updated it to "darkgrey" :-)

File: src/QuestionsForm.js (hunk 6 above)
Comment (carolinwagner) [reply src/QuestionsForm.js:22]: Yes, the "React.Fragment" is needed, because the "<>" doesn't support keys

**Comment by badabam** [comment]
Visual Feedback:
- Your margin on the body is a bit big with 50px. 
- Spacing between Checkboxes is a bit small, give it more space.
- Question 9: Input cuts of the numbers, if they are 3 digits long.
- Text of button at the bottom is way too small. Give it a font-size of 1em.

**Comment by karstenlenz** [comment]
QA results:

The app already works pretty well! I found some possible improvements: 
- on radio buttons clicking the label won't select the value. For checkboxes this works correctly.
- The font-size on the submit-button "Zusammenfassung anzeigen" is pretty small (11px). I wouldn't recommended going below 14px.
- Input elements such as checkboxes or radio buttons are quite small and spaced together closely. This makes it less touch-friendly. Try  a little bit more space in between.
- minor typo: "zweidrittel Mehrheit" should be "Zweidrittelmehrheit" or Zweidrittel-Mehrheit"

**Comment by annidrumm** [comment]
If the text from the input is very long, then it runs outside the screen in the preview below. Maybe you could add a width or break somewhere to prevent it.

See at "Bitte überprüfe nochmal deine Antworten":

Screen Shot 2020-09-06 at 22 17 42


Screen Shot 2020-09-06 at 22 17 30

File: public/index.html (hunk 1 above)
Comment (carolinwagner) [reply public/index.html]: Done

File: src/GlobalStyles.js (hunk 5 above)
Comment (carolinwagner) [reply src/GlobalStyles.js]: Done

**Comment by carolinwagner** [comment]
Thank you for your Feedback @badabam. I implemented your suggestions but the third point 
>     * Question 9: Input cuts of the numbers, if they are 3 digits long.

Unfortunately I cannot reproduce this problem, as it looks like this in my browser: 

Screenshot 2020-09-07 at 10 32 41

Could you please explain what you mean?

**Comment by badabam** [comment]
On iPhone 6/7/8 the input text is cut off:

image

**Pull request merged**